Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -10,39 +10,100 @@ import BraveRewards
 private let logger = Logger.browserLogger
 
 protocol NTPDownloaderDelegate: class {
-    func onNTPUpdated(ntpInfo: NTPBackgroundDataSource.Sponsor?)
+    func onSponsorUpdated(sponsor: NTPSponsor?)
+    func onThemeUpdated(theme: CustomTheme?)
 }
 
 class NTPDownloader {
-    private static let etagFile = "crc.etag"
-    private static let metadataFile = "photo.json"
-    private static let ntpDownloadsFolder = "NTPDownloads"
-    private static let baseURL = { () -> String in
-        switch BraveLedger.environment {
-        case .production: return "https://mobile-data.s3.brave.com/"
-        case .staging, .development: return "https://brave-ntp-crx-input-dev.s3-us-west-2.amazonaws.com/"
-        @unknown default:
-            return "https://mobile-data.s3.brave.com/"
+    enum ResourceType {
+        /// Downloaded only when users installs the app via special referral code.
+        case superReferral(code: String)
+        /// Downloaded for all users if a sponsor is available in their region.
+        case sponsor
+        
+        var resourceBaseURL: URL? {
+            let baseUrl = AppConstants.buildChannel.isPublic ? "https://mobile-data.s3.brave.com/"
+                : "https://brave-ntp-crx-input-dev.s3-us-west-2.amazonaws.com/"
```

Comment:
FYI: this is an open bucket and anyone can look at artifacts within the bucket. A curious user can figure out the new referrers in the pipeline. If we need to keep the new referrers to be private we can think of including these base URLs as part of the build config.

Reply:
Good catch. We also need to request a DNS alias for that bucket from https://github.com/brave/devops/issues/new/choose as we no longer allow Amazon URLs in code or configuration files.

Reply:
Does this apply to NTP SI as well?

Reply:
Yes, our new policy, which is aimed at preventing bucket/Heroku takeovers, is to never have Amazon/Heroku hostnames in shipping products or in revision control.

Reply:
thanks, requested it here

https://github.com/brave/devops/issues/2877